### PR TITLE
Added Prometheus auth proxy

### PR DESCRIPTION
**Pull request by skabashnyuk**
Added Prometheus auth proxy
for https://github.com/redhat-developer/che-monitoring/issues/4

### CODE REVIEW

**Review by aditya-konarde — CHANGES_REQUESTED** [review]
You need a secrets reference in the pod template: 
```
    secrets:
    - app-sre-observability-htpasswd
```

File: openshift/che-monitoring.yaml [hunk 1]
```diff
@@ -63,6 +63,48 @@ objects:
               memory: 256Mi
             requests:
               memory: 128Mi
+        - args:
+          - -provider=github
+          - -http-address=:3000
+          - -https-address=
+          - -htpasswd-file=/etc/proxy/htpasswd/auth
+          - -email-domain=*
+          - -upstream=${PROMETHEUS_SVC_URL}
```
Comment (aditya-konarde) [inline openshift/che-monitoring.yaml:72]: You either need to add this as a template parameter.

Declaration it in the parameters section. 

And then set this value in the saas file once this PR is merged: https://github.com/openshiftio/saas-openshiftio/pull/1283

**Review by aditya-konarde — CHANGES_REQUESTED** [review]

**Comment by aditya-konarde** [comment]
We cannot hard-code this value as the SVC URL's are different for staging and prod.

**Review by ibuziuk — APPROVED** [review]

**Review by aditya-konarde — APPROVED** [review]
LGTM

**Pull request merged**